crates/services/producer/src/block_producer/tests.rs

```diff
@@ -138,6 +142,66 @@ async fn can_produce_next_block() {
     assert!(result.is_ok());
 }
 
+#[tokio::test]
+async fn can_produce_next_block_that_contains_non_zero_versions() {
```

Comment:
We should include the SUT in the test name IMO: `produce_and_execute_block_txpool`.

Also, this test is probably testing too much. We should probably split this into a test for param version and state transition version.

Reply:
I also don't really understand why we should care that the version is non-zero... or why we should assume `rng.gen()` is non-zero for that matter.